daml-assistant/src/DA/Daml/Assistant/Install.hs

```diff
@@ -310,31 +336,38 @@ extractAndInstall env source =
     where throwError msg e = liftIO $ throwIO $ assistantErrorBecause ("Invalid SDK release: " <> msg) e
 
 -- | Download an sdk tarball and install it.
-httpInstall :: InstallEnv -> SdkVersion -> IO ()
-httpInstall env@InstallEnv{..} releaseVersion = do
+httpInstall :: InstallEnvWithVersion -> IO ()
+httpInstall env@InstallEnv{targetVersionM = releaseVersion, ..} = do
     unlessQuiet env $ output "Downloading SDK release."
     requiredAny "Failed to download SDK release." $ do
-        sdkVersion <- Github.getSdkVersionFromEnterpriseVersion releaseVersion
-        downloadLocation $ getLocation sdkVersion
+        downloadLocation =<< getLocation releaseVersion
     where
-        getLocation :: SdkVersion -> InstallLocation
-        getLocation sdkVersion = case artifactoryApiKeyM of
-            Nothing -> Github.versionLocation releaseVersion sdkVersion
-            Just apiKey
-              | releaseVersion >= firstEEVersion -> Artifactory.versionLocation sdkVersion apiKey
-              | otherwise -> Github.versionLocation releaseVersion sdkVersion
+        getLocation :: ReleaseVersion -> IO DAVersion.InstallLocation
+        getLocation releaseVersion = do
+            damlConfigE <- tryConfig (readDamlConfig damlPath)
+            let alternateDownload =
+                    join $ eitherToMaybe $ queryDamlConfig ["alternate-download"] =<< damlConfigE
+            pure $ case alternateDownload of
+              Just url -> DAVersion.alternateVersionLocation releaseVersion url
+              Nothing ->
+                case artifactoryApiKeyM of
+                    Nothing -> DAVersion.githubVersionLocation releaseVersion
+                    Just apiKey
+                        -- TODO: Define ordering over release versions
```

Comment:
I'd say compare Split's release version to old version
so ```compare `on` releaseVersionFromReleaseVersion```
We should always have release version unique, regardless of sdk version or new vs old, so no need to order deeper than that

Reply:
fixed